docs/src/index.md

```diff
@@ -118,6 +115,68 @@ then I can use the pushforward to find ``\dfrac{∂f}{∂x}``
 
 ``\dfrac{∂f}{∂x}=\mathrm{pushforward}_{h(b)|b=g(x)}\left(\left.\dfrac{∂g}{∂a}\right|_{a=x}\right)``
 
+##### Geometric interpretation of reverse and forwards mode AD
+
+Let us think of our types geometrically. In other words, elements of a type form a _manifold_.
+This document will explain this point of view in some detail.
+
+###### Some terminology/conventions.
+
+Let ``p`` be an element of type M, which is defined by some assignment of numbers ``x_1,...,x_m``,
+say ``(x_1,...,x_m) = (a_1,...,1_m)`` 
+
+A _function_ ``f:M -> K`` on ``M`` is (for simplicity) a polynomial ``K[x_1, ... x_m]``
+
+The tangent space ``T_pM`` of ``T`` is the ``K``-vector space spanned by derivations ``d/dx``. 
+The tangent space acts linearly on the space of functions. They act as usual on functions. Our starting point is 
```

Comment:
Curves are a nice interpretation of tangent vectors, but I'm not convinced they are the right one for AD, since I don't think they necessarily add anything. 

For reference, we can define an isomorphism 
maps from the first order neighborhood of 0 in ``K`` (infinitesimal curves)  -> tangent space at ``p`` 
by considering the derivation in the direction of the curve.